File: lab2/lab2_1.cpp
#include <stdio.h>
#include <ctime>

class TimeLogger {
public:
  TimeLogger() {
    time_t now = time(0);
    tm *ltm = localtime(&now);
    printf("Object created on: %d/%d/%d %d:%d:%d\n",
            ltm->tm_mday, 1 + ltm->tm_mon, 1900 + ltm->tm_year,
            ltm->tm_hour, ltm->tm_min, ltm->tm_sec);

    int h = 23 - ltm->tm_hour;
    int m = 59 - ltm->tm_min;
    int s = 59 - ltm->tm_sec;
    printf("Time left until the end of the day: %d hours, %d minutes, %d seconds\n", h, m, s);
  }
};

int main() {
  TimeLogger obj;
  return 0;
}

File: lab2/lab2_1.kt
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

class TimeLogger {
  init {
    val now = LocalDateTime.now()
    val formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss")
    println("Object created on: ${now.format(formatter)}")

    val h: Int = 23 - now.hour
    val m: Int = 59 - now.minute
    val s: Int = 59 - now.second
    println("Time left until the end of the day: $h hours, $m minutes, $s seconds")
  }
}

fun main() {
  val obj = TimeLogger()
}